Hackney.Shared.Finance/Boundary/Response/ResponseWrapper.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Hackney.Shared.Finance.Boundary.Response
+{
+    public class ResponseWrapper<T> where T : class
+    {
+        public T Value { get; set; }
+
+        public bool IsEmpty
+        {
+            get
+            {
+                if (this.Value is IEnumerable<object>)
+                {
+                    return !(this.Value as IEnumerable<object>).Any();
+                }
```

Comment:
Nice code, though, is there any case where we'd want to do such check before returning an IActionResult response?

From what I remember from REST, upon querying an existing collection _(say Assets)_ with filter parameters, `even if no results are found`, the standard expected endpoint response is `200 Ok` with an empty collection: `[ ]`. Which represents the fact that collection exists, but it's either empty, or it contains nothing matching your filter parameters.

The only time the `GET multiple things` endpoint is expected to receive a `404 Not Found` back is when the collection itself doesn't exist - so we don't even have the collection name specified to even query against.

So, for the above reasons, I believe that there is no circumstance, where we'd want to this **Line 14-17** check.

Reply:
This code does have a use case in the Transactions-api.

In the pattern we have used, IEnumerable<T> collections are returned from the Gateway (Repository) layer for those methods that are expected to return a collection of data. An IEnumerble.Any() (or better still IEnumerable.Empty<T>) check is still checking on an IEnumerable. 

If you have a method which returns an IEnumerable<T> and the internal code of that method has found no records, then it is far more desirable to return an IEnumable.Empty<T> than a null. 

An empty collection is far more representative of a 404 Not Found than null.